src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,212 @@
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert car to DB. Car: " + car, e);
+        }
+        insertDrivers(car);
+        return car;
+    }
+
+    private void insertDrivers(Car car) {
+        String insertDriversRequest = "INSERT INTO cars_drivers (car_id, driver_id) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement addDriversToCarStatement =
+                        connection.prepareStatement(insertDriversRequest)) {
+            addDriversToCarStatement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                addDriversToCarStatement.setLong(2, driver.getId());
+                addDriversToCarStatement.executeUpdate();
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't insert drivers to car. Car: " + car, e);
+        }
+    }
+
+    @Override
+    public Optional<Car> get(Long id) {
+        String selectRequest = "SELECT c.id as car_id, c.model as car_model, m.id as"
+                + " manufacturer_id, m.name as manufacturer_name, m.country as "
+                + "manufacturer_country FROM cars c JOIN manufacturers m ON "
+                + "c.manufacturer_id = m.id  where c.id = ? and c.is_deleted = false;";
+        Car car = null;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getCarStatement =
+                        connection.prepareStatement(selectRequest)) {
+            getCarStatement.setLong(1, id);
+            ResultSet resultSet = getCarStatement.executeQuery();
+            if (resultSet.next()) {
+                car = parseCarWithManufacturerFromResultSet(resultSet);
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't find car in DB by id: " + id, e);
+        }
+        if (car != null) {
+            car.setDrivers(getDriversForCar(id));
+        }
+        return Optional.ofNullable(car);
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String selectAllRequest = "SELECT c.id as car_id, c.model as car_model, m.id as"
+                + " manufacturer_id, m.name as manufacturer_name, m.country as "
+                + "manufacturer_country FROM cars c JOIN manufacturers m ON "
+                + "c.manufacturer_id = m.id where c.is_deleted = false;";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllCarsStatement =
+                        connection.prepareStatement(selectAllRequest)) {
+            ResultSet resultSet = getAllCarsStatement.executeQuery();
+            while (resultSet.next()) {
+                Car car = parseCarWithManufacturerFromResultSet(resultSet);
+                car.setDrivers(getDriversForCar(car.getId()));

```

Comment:
You have to close the PreparedStatement after you’re done with it and before you create a new one on the same connection.